Documentation/gettingstarted/k8s-install-openshift-okd.rst

```diff
@@ -0,0 +1,202 @@
+
+.. _k8s_install_openshift_okd:
+
+*****************************
+Installation on OpenShift OKD
+*****************************
+
+OpenShift Requirements
+======================
+
+1. Choose preferred cloud provider. This guide was tested in AWS, Azure & GCP.
+
+2. Read `OpenShift documentation <https://docs.okd.io/latest/welcome/index.html>`_ to find out about provider-specific prerequisites
+
+3. `Get OpenShift Installer <https://github.com/openshift/okd#getting-started>`_
+
+.. note:: It's highly recommended to read the docs, unless you have installed
+   OpenShift in the past. Here are a few notes that you may find useful.
+   
+   - with the AWS provider ``openshift-install`` will not work properly
+     when MFA credentials are stored in ``~/.aws/credentials``, traditional credentials are required
+   - with the Azure provider ``openshift-install`` will prompt for
+     credentials and store them in ``~/.azure/osServicePrincipal.json``, it
+     doesn't simply pickup ``az login`` credentials. It's recommended to
+     setup a dedicated service principal and use it
+   - with the GCP provider ``openshift-install`` will only work with a service
+     account key, which has to be set using ``GOOGLE_APPLICATION_CREDENTIALS``
+     environment variable (e.g. ``GOOGLE_APPLICATION_CREDENTIALS=service-account.json``)
+
+Create an OpenShift OKD Cluster
+===============================
+
+First, set cluster name:
+
+.. code:: bash
+
+   CLUSTER_NAME="cluster-1"
+
+Now, create configuration files:
+
+   NOTE: the sample output below is showing the AWS provider, but
+   it should work the same way with other providers
+
+.. code:: shell-session
+
+   $ openshift-install create install-config --dir "${CLUSTER_NAME}"
+   ? SSH Public Key ~/.ssh/id_rsa.pub
+   ? Platform aws
+   INFO Credentials loaded from default AWS environment variables 
+   ? Region eu-west-1
+   ? Base Domain openshift-test-1.cilium.rocks
+   ? Cluster Name cluster-1
+   ? Pull Secret [? for help] ******************************************************************************************************************************************************************************************
+
+And set ``networkType: Cilium``:
+
+.. code:: bash
+
+   sed -i 's/networkType:\ OVNKubernetes/networkType:\ Cilium/' "${CLUSTER_NAME}/install-config.yaml"
+
+Next, generate OpenShift manifests:
+
+.. code:: bash
+
+   openshift-install create manifests --dir "${CLUSTER_NAME}"
+
+Now, define ``cilium`` namespace:
+
+.. code:: bash
+
+   cat << EOF > "${CLUSTER_NAME}/manifests/cluster-network-03-cilium-namespace.yaml"
+   apiVersion: v1
+   kind: Namespace
+   metadata:
+     name: cilium
+     annotations:
+       # node selector is required to make cilium-operator run on control plane nodes
+       openshift.io/node-selector: ""
+     labels:
+       name: cilium
+       # run level sets priority for Cilium to be deployed prior to other components
+       openshift.io/run-level: "0"
```

Comment:
@joestringer I am going to have a look at various annotations and labels used e.g. for OVN and see if there is anything we should add here.

Reply:
Here is what Multus namespace has:
```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    openshift.io/description: Multus network plugin components
    openshift.io/node-selector: ""
    openshift.io/sa.scc.mcs: s0:c10,c0
    openshift.io/sa.scc.supplemental-groups: 1000090000/10000
    openshift.io/sa.scc.uid-range: 1000090000/10000
  creationTimestamp: "2020-10-05T15:32:22Z"
  labels:
    name: openshift-multus
    olm.operatorgroup.uid/094d3245-36c5-470a-9d31-c73eb2ea4892: ""
    openshift.io/cluster-monitoring: "true"
    openshift.io/run-level: "0"
   name: multus
```

Reply:
https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/

Reply:
🤔 https://github.com/cilium/cilium/blob/503b073b7a7c948329e581eb0c49c6cde37ae6a4/install/kubernetes/cilium/charts/agent/templates/daemonset.yaml#L367-L369

Reply:
Ok, so the template logic is actually fine, however I can see that it's not working for this guide due to the use of `helm template`.

Reply:
It looks like in my first cluster `priorityClassName` wasn't set, but it did worked in another two clusters I just created. It's rather odd, I will have to look into why would `helm template` behave like this.

Reply:
Ok, turns out what changed things is Helm upgrade that I did 'just in case' while testing today.

I opened another issue here: https://github.com/cilium/cilium/issues/13428.

Reply:
I added `openshift.io/cluster-monitoring` and `openshift.io/cluster-logging` labels, but not so sure about `openshift.io/sa.scc.*` annotations, we may need these, but doesn't look like it's neccessary.

Reply:
@joestringer I think we can close this thread, the PR should be satisfactory as it stands. We do need to deal with #13428 separately.